contents/node/nodejs.md

```diff
@@ -0,0 +1,209 @@
+
+There are also useful non-core modules maintained by the community and external developers such as *Bluebird* for promises.
+
+Importing modules is simple - just use the `require()` function and provide the module identifier or the file path.
+```js
+const http = require('http'); // import a core module
+const Promise = require('bluebird'); // import a non-core module
+```
+
+Here is a brief description of how Node locates a module.
+
+Node will first check if the module identifier passed to `require` is a core module or a relative path. If so, it will return the core module or the object which references the value of `module.exports` in the specified file path respectively. If this check fails, Node will start at the parent directory of the current module and appends `/node_modules`, and attempts to load the module from that location. If the module still can't be found, it continues walking up the parent directory (and repeating the step) until the root of the file system is reached.
+
+### Organizing code
+
+We can also create separate files and modules in our codebase such that each module is focused on a single functionality.
+
+As an example, in `parser.js`,
+```js
+function Parser(options) {
+    this._options = options || {};
+}
+
+Parser.prototype.parse = function (content) {
+  // insert code here
+}
+module.exports = Parser; //override the exports object
+```
+The Parser function/constructor is returned whenever the `require` function is used to include the `Parser` module.
+
+In a separate file in the project,
+```js
+const parser = require('../parser') // import content from parser.js based on relative file path
+
+const content = 'some content';
+parser.parse(content);
+```
+Both relative and absolute file paths are accepted. Note that the `js` extension is assumed and can be excluded.
+
+To understand how to use `module.exports`, checkout this [article by Tendai Mutunhire](http://stackabuse.com/how-to-use-module-exports-in-node-js/)
+
+# Writing a Hello World Node.js server
+
+Here is example on how to write a Hello World Node.js program taken from a [codeburst article](https://codeburst.io/getting-started-with-node-js-a-beginners-guide-b03e25bca71b).
+
+It demonstrates how a Node.js application import required modules, create a server to listen to client's request, read the request and returns a response.
+
+Create a file `server.js` with following content:
+
+```js
+const http = require('http');
+
+const hostname = '127.0.0.1';
+const port = 3000;
+
+const server = http.createServer(function(req, res) {
+  res.statusCode = 200;
+  res.setHeader('Content-Type', 'text/plain');
+  res.end('Hello World');
+});
+server.listen(port, hostname, function() {
+  console.log('Server running at http://'+ hostname + ':' + port + '/');
+});
+```
+
+After you have saved the file, you can execute it from your terminal like so:
+```
+$ node helloworld.js
```

Comment:
Shouldn't this be `node server.js` ?

Reply:
Should we standardize place a `$` for all terminal commands?